### PR TITLE
Privacy policy

### PR DESCRIPTION
**Pull request by NoahMcGraw**
Phase 2- Add the rest of the questionnaire and the summary page. Policy can now be completely edited and generated/printed. 

Notes: 
Discussed printing method did not work as expected. Attempted to create a workaround but ran into issues.. Position: fixed would not display the full contents of the div on print. position: absolute Did display the contents of the div but could not gain full use of the window Because it was the child of a div that was separated from the banner and rest of the app and therefore could only, at the most, occupy the full dimensions of the parent div. Would not be an issue since we could add .:not(.generatedPolicy) to set everything else to display: none, EXCEPT that this also applies to the children of generatedPolicy and specifying .generatedPolicy > * in the :not does not seem to be effective.

Regardless, I was actually able to get the previous print function to work across browsers(Including Chrome) with relative ease AND I think it is a much cleaner way to do this anyway so I will include this version in the push.

A phone number mask has been added. However, the mask does not work very well with ng-input and will clash with the labels that already exist.

### CODE REVIEW

File: app/controllers/main-ctrl.js [hunk 1]
```diff
@@ -64,4 +82,53 @@ app.controller('mainCtrl', ['$scope', '$mdDialog', function($scope, $mdDialog){
...
     }
+
+    $scope.printElem = function(elem) {
+        $scope.newPolicy.printing = true;
+    // Initialize variables //
+        var html = "";
+        var head = "";
+        var body = "";
+        var title = "<title>" + $scope.newPolicy.companyName + " Privacy Policy</title>";
+        var angularMaterialCss = '<link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/angular_material/1.1.8/angular-material.min.css">';
```
Comment (reesemartin) [inline app/controllers/main-ctrl.js:93]: Might also be a good idea to include the app.css to ensure any global default styles of your app are included.

File: app/partials/privacyPolicyGenerator-template.html [hunk 2]
```diff
@@ -9,17 +9,18 @@
...
                 <p>Let us do the hard work for you! With our free <strong>Privacy Policy Generator</strong>,
                 <br>
                 you can have a custom policy on your desk in as little as <strong>Five Minutes</strong>!
                 </p>
             </div>
-            <md-button class="md-primary md-raised unisyn-big-button" ng-click="startNewPolicy()" ng-hide="newPolicy.addingInfo || newPolicy.questionaire">Let's Get Started</md-button>
+            <md-button class="md-primary md-raised unisyn-big-button" ng-click="startNewPolicy()" ng-show="newPolicy.untouched">Let's Get Started</md-button>
         </md-content>
     </div>
     <div layout="column" layout-align="center center">
```
Comment (reesemartin) [inline app/partials/privacyPolicyGenerator-template.html]: Here is one the clear issues on mobile.

layout directives should not be used when you are using pixels to restrict its children. The point of the directive is that you make it into a "table" and the "cells" are portions that add up to 100%. A rare exception would be if you need vertical centering as well. But here you only appear to need horizontal centering.

If all you care about is that its content is centered then move the privacyPolicyGeneratorContainer class to this element as the actual policy generator contain and then set its width as described in the overall PR review and then give it:
```
margin: auto;
```
which will center horizontally.

File: app/partials/questionnaireTabs/questionnaireUsage-template.html [hunk 3]
```diff
@@ -51,12 +51,19 @@
...
-<div ng-show="questionnaire.usage.answer">
-    <div class="md-title">Please provide a brief description</div>
-    <md-input-container class="md-block" flex>
-        <label>Description</label>
-        <textarea ng-model="questionnaire.usage.answerDescription"></textarea>
-    </md-input-container>
+<div ng-show="questionnaire.usage.answer" class="md-vertical-padding-bottom">
+    <div class="md-title">Please list the additional reasons you will access user data</div>
+    <md-chips ng-model="questionnaire.usage.answerDescription"
+              readonly="false"
```
Comment (reesemartin) [inline app/partials/questionnaireTabs/questionnaireUsage-template.html]: This doesnt need to be readonly="false" since false is the default if you dont even include this attribute.

File: app/partials/questionnaireTabs/questionnaireStorage-template.html [hunk 4]
```diff
@@ -59,10 +59,19 @@
...
-    <md-input-container class="md-block" flex>
-        <label>Description</label>
-        <textarea ng-model="questionnaire.storage.answerDescription"></textarea>
-    </md-input-container>
+<div ng-show="questionnaire.storage.answer" class="md-vertical-padding-bottom">
+    <div class="md-title">
+        Please list any additional data that will be tracked and stored
+    </div>
+    <md-chips ng-model="questionnaire.storage.answerDescription"
+              readonly="false"
```
Comment (reesemartin) [inline app/partials/questionnaireTabs/questionnaireStorage-template.html]: This doesnt need to be readonly="false" since false is the default if you dont even include this attribute.

File: app/partials/questionnaireTabs/questionnaireSharing-template.html [hunk 5]
```diff
@@ -28,23 +28,32 @@
...
-    <md-input-container class="md-block" flex>
-        <label>Description</label>
-        <textarea ng-model="questionnaire.sharing.answerDescription"></textarea>
-    </md-input-container>
+<div ng-show="questionnaire.sharing.answer" class="md-vertical-padding-bottom">
+    <div class="md-title md-underline">
+        Please list the additional reasons you will share user data
+    </div>
+    <md-chips ng-model="questionnaire.sharing.answerDescription"
+              readonly="false"
```
Comment (reesemartin) [inline app/partials/questionnaireTabs/questionnaireSharing-template.html]: This doesnt need to be readonly="false" since false is the default if you dont even include this attribute.

**Review by reesemartin — CHANGES_REQUESTED** [review]
The app is not very mobile friendly. This is mostly due to the px widths you have defined. See the stylesheet notes I have added to our coding guidelines repo here: https://github.com/UniSynTechnologies/coding-guidelines/blob/master/stylesheets.
Instead of saying width 900px make it width 100% with max-width of 900px. Though I dont think that is even necessary. Set a max width on the app itself and then let all the contents be 100%. That makes your app more modular so that if you drop this into something else, you can control its width / mobile aspects based on the top level vs each thing inside trying to be its own width.

Also with styling I notice you have some scroll bars on mobile due to the padding not being included when doing widths. This is due to the box-sizing css property which most apps set as "border-box". We do this in our reset.css file in TopSale. You can just put a quick rule at the top of app.css like so:
```
privacy-policy-generator * {
    box-sizing: border-box;
}
```

It is a generally a good idea to adjust the width of your browser window whenever you make css changes to make sure they look good on mobile. I try to do a mobile first design methodology where I design the interface with my browser window thin like mobile and then I expand it and add conditions to spread things out for desktop.

This is a good way to catch mobile issues like the ones I described early on before it requires hacking out parts of your code to fix layout.

-------------------------------------------------

The issue with your printing of the policy fullscreen was likely due to the 600px height you have on the policy container and the zindex. The following rule applied to generatedPolicy does work in my dev:
```
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background: #fff;
    z-index: 999999;
    height: initial;
```
But I do see what you have in place and it works too. I dont like that there is html in the controller since the breaks the model / view / controller scheme but its not worth spending the time to change it since it works pretty well. Great job confirming it works in other browsers.

----------------------------------------------------

Final nitpick thing:
All files should have 1 blank line at the end of the file. Lots of discussion about this general "rule" here: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline but it really comes down to being because some text parsers only could lines ending in new line characters are actual lines to parse. So if our code is ever passed through a 3rd party parser for minification or uglification or some other automated process there is a slight chance this could be a problem.

Look at the files changed in this PR and you will see that git actually makes a red notation at the end of your files for the missing new line character.

Try to remember to add new lines at the end of the files. I know its not easy to remember but try your best to.

**Review by reesemartin — APPROVED** [review]
